### PR TITLE
[Port dspace-7_x] Refactored community & collection pages

### LINKED ISSUE #2700: [Accessibility] Browse By Tabs on Community/Collection Homepages are missing ARIA labels.
Originally reported in #1178 as part of the Deque Analysis in 2021.  This was rated as "serious" by Deque (one level down from "critical") which means:
> Serious: This issue results in serious barriers for people with disabilities, and will partially prevent them from accessing fundamental features or content. People relying on assistive technologies will experience significant frustration as a result. Issues falling under this category are major problems, and remediation should be a priority.

**Describe the bug**

This issue impacts **both** the Community and Collection homepages: 
* (ID 469552, 469655) "Selected state of the element is missing or incorrect." on the currently selected browse by option (e.g. "Subcommunities and Collections" on Community homepage, "Recent Submission" on Collection homepage). In both cases, the selected option is displayed visually, but not noted in ARIA tags.
    * Fix: Use the `aria-current` attribute to indicate the element that represents the current item within a container or set of related elements, such as the currently selected page, step, location, date, or time. https://www.w3.org/TR/wai-aria-1.1/#aria-current

**To Reproduce**
1. Can be reproduced on the demo site with these sample communities/collections:
    * https://demo.dspace.org/communities/0958c910-2037-42a9-81c7-dca80e3892b4
    * https://demo.dspace.org/collections/282164f5-d325-4740-8dd1-fa4d6d3e7200
2. When on either of those pages, verify that the Browse By tabs (under Collection/Community page title) do not have `aria-current` attributes.

**Related work**
Originally reported in #1178

### PR DESCRIPTION
**Pull request by alexandrevryghem**
Partial port of #2339 & #2722 by @alexandrevryghem  to dspace-7_x.

Contains:
- Dynamic component loader performance fix
- Fixes #2700

### CODE REVIEW

**Comment by tdonohue** [comment]
Thanks @alexandrevryghem .  I'll give this a final review/test once the associated 8.0 PRs are both merged.  That way we can make sure to update this if any additional modifications need to be backported.

File: src/app/shared/comcol/comcol-page-browse-by/comcol-page-browse-by.component.html [hunk 1]
```diff
@@ -2,9 +2,11 @@ <h2 class="comcol-browse-label h5">{{'browse.comcol.head' | translate}}</h2>
 <nav class="comcol-browse mb-4" aria-label="Browse Community or Collection">
   <div class="d-none d-sm-block">
 
-    <div class="list-group list-group-horizontal">
+    <div class="list-group list-group-horizontal" role="tablist">
       <a *ngFor="let option of allOptions"
+         [attr.aria-current]="(currentOptionId$ | async) === option.id"
```
Comment (tdonohue) [inline src/app/shared/comcol/comcol-page-browse-by/comcol-page-browse-by.component.html:7]: Strangely, this doesn't entirely work on the `dspace-7_x` branch.  

When the tab is NOT enabled, I *do* see `aria-current=false`.  
But, when a tab *is enabled*, the `aria-current` attribute **disappears**. On the main branch, I see `aria-current=true`.

**Review by tdonohue — COMMENTED** [review]
@alexandrevryghem : I tested this backport PR directly, just to verify the fixes are working on `dspace-7_x`.  Overall, it looks good.  However, I noticed a *very minor* `aria-current` issue with the tabs on Community/Collection pages.  See inline comment below.

Overall, I'm +1 this backport.  But, it'd be good to fix that ARIA issue if possible.

**Comment by alexandrevryghem** [comment]
@tdonohue: Thnx! This issue was cause by the `routerLinkActive="active"` attribute on that link. Replacing it with `[class.active]="(currentOptionId$ | async) === option.id"` like on `main` fixed this issue

**Review by tdonohue — APPROVED** [review]
👍 Thanks @alexandrevryghem !  Verified this now works the same as the 8.x version.  Merging this backport PR immediately.

**Pull request merged**